src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -69,12 +87,63 @@ RouterExecutorStart(QueryDesc *queryDesc, int eflags, Task *task)
 	/* ensure that the task is not NULL */
 	Assert(task != NULL);
 
-	/* disallow transactions and triggers during distributed modify commands */
+	/* disallow triggers during distributed modify commands */
 	if (commandType != CMD_SELECT)
 	{
-		bool topLevel = true;
-		PreventTransactionChain(topLevel, "distributed commands");
 		eflags |= EXEC_FLAG_SKIP_TRIGGERS;
+
+		if (IsTransactionBlock() && xactParticipantHash == NULL)
```

Comment:
(From Hangouts: PLs will inevitably use the non-remote destination, otherwise they won't get the result, so this solution covers all functions. Forget about multi-command strings.)